doniclient/osc/cli.py

```diff
@@ -115,246 +74,226 @@ def take_action(self, parsed_args):
-                f"\nOptional Keys are {self.iface_optional_keys}"
-            ),
-        )
-        update_keys = ["index"]
-        update_keys.extend(self.iface_required_keys)
-        parse_iface.add_argument(
-            "--update",
-            required_keys=update_keys,
-            optional_keys=self.iface_optional_keys,
-            action=parseractions.MultiKeyValueAction,
-            help=(
-                "Specify once per interface, in the form:"
-                "`--interface key1=value1, key2=value2, ...`"
-                f"\nRequired Keys are {update_keys}"
-                f"\nOptional Keys are {self.iface_optional_keys}"
-            ),
-        )
-        parse_iface.add_argument(
-            "--delete",
-            help=("Specify interface to delete, by index`"),
-        )
+        args_to_default = ("name", "hardware_type", "properties")
+        # Unset all defaults to avoid accidental changes
+        for arg in parser._get_optional_actions():
+            if arg.dest in args_to_default:
+                arg.default = argparse.SUPPRESS
 
         return parser
 
-    def _handle_ifaces(self, parsed_args):
-        # Update Interfaces
-        patch = []
-        for iface in getattr(parsed_args, "add") or []:
-            patch.append(
-                {"op": "add", "path": f"/properties/interfaces/-", "value": iface}
-            )
-
-        for iface in getattr(parsed_args, "update") or []:
-            index = iface.pop("index")
-            patch.append(
-                {
-                    "op": "replace",
-                    "path": f"/properties/interfaces/{index}",
-                    "value": iface,
-                }
-            )
-        for index in getattr(parsed_args, "delete") or []:
-            patch.append({"op": "remove", "path": f"/properties/interfaces/{index}"})
-        return patch
-
     def get_patch(self, parsed_args):
         patch = []
+
         field_map = {
             "name": "name",
             "hardware_type": "hardware_type",
-            "management_address": "properties/management_address",
-            "ipmi_username": "properties/ipmi_username",
-            "ipmi_password": "properties/ipmi_password",
-            "ipmi_terminal_port": "properties/ipmi_terminal_port",
-            "deploy_kernel": "properties/baremetal_deploy_kernel_image",
-            "deploy_ramdisk": "properties/baremetal_deploy_ramdisk_image",
         }
+
         for key, val in field_map.items():
             arg = getattr(parsed_args, key, None)
             if arg:
                 patch.append({"op": "add", "path": f"/{val}", "value": arg})
 
-        subparser = getattr(parsed_args, "func", None)
-        if subparser:
-            patch.extend(subparser(parsed_args))
+        try:
+            for key, val in parsed_args.properties.items():
+                patch.append({"op": "add", "path": f"/properties/{key}", "value": val})
```

Comment:
I can see this pattern was used before, but I'm wondering if `"add"` will work when the operation is really a `"replace"`, does doni gracefully handle it?